Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1418,11 +1418,6 @@ void Document::update_layout(UpdateLayoutReason reason)
         return TraversalDecision::Continue;
     });
 
-    // Scrolling by zero offset will clamp scroll offset back to valid range if it was out of bounds
-    // after the viewport size change.
-    if (auto window = this->window())
-        window->scroll_by(0, 0);
```

Comment:
This breaks resizing the viewport vs. allowed scroll positions. For example, go to https://ladybird.org, make the viewport very narrow, scroll all the way down, and make the viewport wide again.

Reply:
We should probably have a test for this 🤓